verifier.py

```diff
@@ -417,73 +379,50 @@ def main():
-                    for url in run_info['node_urls']:
-                        if not url.endswith("fail"):
-                            run_info['node_urls'].remove(url)
-                    rescrape_list.extend(project_dashboards.failed_pages)
+                    project_dashboards = ProjectDashboardVerifier().failed_pages
+                    nodes_list_verified += project_dashboards
             if run_info['scrape_registrations']:
                 # Must run all page types automatically
-                registration_files = RegistrationFilesVerifier()
-                registration_wiki = RegistrationWikiVerifier()
-                registration_analytics = RegistrationAnalyticsVerifier()
-                registration_forks = RegistrationForksVerifier()
-                registration_dashboards = RegistrationDashboardVerifier()
-                for url in run_info['registration_urls']:
-                    if not url.endswith("fail"):
-                        run_info['registration_urls'].remove(url)
-                rescrape_list.extend((registration_files.failed_pages + registration_wiki.failed_pages +
-                                      registration_analytics.failed_pages + registration_forks.failed_pages +
-                                      registration_dashboards.failed_pages))
+                registration_files = RegistrationFilesVerifier().failed_pages
+                registration_wiki = RegistrationWikiVerifier().failed_pages
+                registration_analytics = RegistrationAnalyticsVerifier().failed_pages
+                registration_forks = RegistrationForksVerifier().failed_pages
+                registration_dashboards = RegistrationDashboardVerifier().failed_pages
+                registrations_list_verified = registration_files + registration_wiki + registration_analytics + \
+                    registration_forks + registration_dashboards
             if run_info['scrape_users']:
-                user_profiles = UserProfileVerifier()
-                for url in run_info['user_profile_page_urls']:
-                    if not url.endswith("fail"):
-                        run_info['user_profile_page_urls'].remove(url)
-                rescrape_list.extend(user_profiles.failed_pages)
+                user_profiles_verified = UserProfileVerifier().failed_pages
             if run_info['scrape_institutions']:
-                institution_dashboards = InstitutionDashboardVerifier()
-                for url in run_info['institution_urls']:
-                    if not url.endswith("fail"):
-                        run_info['institution_urls'].remove(url)
-                rescrape_list.extend(institution_dashboards.failed_pages)
-    # json.dump[run_info]   ??????
+                institution_dashboards_verified = InstitutionDashboardVerifier().failed_pages
 
-    # Rescraping
+            # clearing json file and dumping in new lists
+            run_info['node_urls'] = nodes_list_verified
```

Comment:
I think these assignments need to be inside the if statements. The problem was that there never was an institution_dashboards_verified because it would have been initialized inside the `if run_info['scrape_institutions']` block